### DISCUSSION TITLE
STT concurrent requests limit for a Language-Learning-App

### DISCUSSION THREAD

**Discussion by Simonuhl** [deepgram/community#845] [General help]
Hi, I was send here by Danny Kim (Product Consultant, Deepgram). 

We are developing a language-learning-App where the user builds sentences, either by tapping the displayed words, typing via the keyboard, or speaking in the sentences. 

For the speaking-in-function we are thinking about using deepgrams Nova-2 STT service, without any audio intelligence functions. We have heard very good things.

In the Pay-As-You-Go and the Growth-Plan there is a concurrent requests limit of 100 requests. In the Enterprise-Plan no limit is mentioned.

If the user of our App is choosing the speaking-in-function there will be many short STT requests who need to be handled in real time. It could be that a user will send STT requests of about 5 seconds length, every 20 seconds (upper end case). 

Depending on our success we could reach a situation soon where we have more than 100 users sending requests at the same time, maybe many more. 

Our question is: Is it possible to use Deepgram (Nova-2 STT no audio intelligence) for our language-learning-App? How can we get around the 100 concurrent requests limit?

We want to give our users full autonomy on how they build the sentences, including the speaking-in-function, how many sentences they build and at what time. 

Thank you very much,
all the best
Simon

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
Regarding this question:

> Is it possible to use Deepgram (Nova-2 STT no audio intelligence) for our language-learning-App? How can we get around the 100 concurrent requests limit?

Are you asking if this is possible if you are on a Pay-As-You-Go or a Growth plan? On an Enterprise plan, this is possible, but on the other plans you would hit the requests limit of 100 requests. 

In that situation, you would need to consider a programmatic fix, such as implementing a queue system on your end to manage the requests sent to Deepgram's API. This way, you can control the rate at which requests are sent and avoid hitting the limit. 

Or you could implement a retry mechanism. If the number of in-progress requests for a given project is equal to or greater than the concurrency limit, a new request will receive a 429 response. In case of receiving a 429 response, you could implement a retry mechanism with exponential backoff, to ensure that requests that do get rate-limited eventually succeed. We have a [related blog post](https://deepgram.com/learn/api-back-off-strategies) on backoff strategies for rate limiting.

**Comment by Simonuhl** [comment]
Hey Sandra,
thanks for the fast reply.

> Are you asking if this is possible if you are on a Pay-As-You-Go or a Growth plan? On an Enterprise plan, this is possible, but on the other plans you would hit the requests limit of 100 requests.

No, I wanted to know weather it is possible to not have the request limit in general. I already read the blog post but it is not possible to put a delay on our side, since it would completely break the user experience of our app. It would essentially render the speaking-in-function useless. 

We are very happy that you offer a version without the request limit. 

We want to set up an Enterprise plan.
How can we do this? 
Should I contact Danny again, who send us here or do we set up something here?

Best regards
Simon

**Reply by SandraRodgers** [reply]
Hi Simon,

I'm glad I was able to answer your question. I am contacting our sales team internally to have someone reach out to you ASAP to discuss the enterprise solution with you. You can definitely message Danny again, but I will also make sure that someone reaches out to you today. And my email is sandra.rodgers@deepgram.com if you have any other questions (although it is good to ask them in the community if they are questions that others might benefit from knowing the answer to). Thanks!

Sandra
